core/trace/src/main/scala/org/typelevel/otel4s/trace/Tracer.scala

```diff
@@ -172,6 +174,8 @@ trait Tracer[F[_]] extends TracerMacro[F] {
     */
   def noopScope[A](fa: F[A]): F[A]
 
+  def mapK[G[_]: Sync](fk: F ~> G, gk: G ~> F): Tracer[G]
```

Comment:
if you need both `F ~> G` and `G ~> F` then actually it's an `imapK` 😜 also sometimes the method is called `translate`.

Also, `Sync` is a very strong constraint, and I don't think it's necessary. Once you have `F ~> G` and `G ~> F` then you should be able to derive `Sync[G]` from `Sync[F]`.

See a similar idea in:
- https://github.com/http4s/http4s/pull/6139

Reply:
`translate` is more suitable indeed. 

> Once you have F ~> G and G ~> F then you should be able to derive Sync[G] from Sync[F].

I didn't know that! I will take a look at the example, looks interesting.

Reply:
If you need F ~> G and G ~> F, then the invariant nature means almost all interesting transformers are eliminated from use, OptionT, EitherT, Kleisli, IorT, etc.

Reply:
Random thought: you might be able to do this with something weaker than `F ~> G` + `G ~> F`, like Haskell's `UnliftIO`. That only gets you `Kleisli` while staying sensible, but it's something at least.

Prototype: https://github.com/Prillan/otel4s/commit/355c68d8edefd08b0bb8f81c8e479c0c479de5c7